### PR TITLE
Added the new vocabulary file for undefined terms

### PR DESCRIPTION
**Pull request by iherman**
This is to settle issue #1534. Some remarks, though:

- The PR also includes a small change on the `issuer-dependent.html` file, namely the VCDM reference has been updated. However, the question is whether this file is still necessary or not or, under the circumstances, should be removed.
- I have made the changes on the `/ns/credentias/.htaccess` file. Because the file is not on github, it cannot be managed directly by the PR, so I copy this verbatim below. Please, give your comments here, and I will make the changes separately.
- @msporny there is a change to be done in the repo management, too (at least I presume). At the moment, `https://w3c.github.io/vc-data-model/contexts/credentials/undefined-term/v2` is a 404; I presume you have some magic with gh-pages going on which must be adapted for the new file. I am not sure what to do with this, I leave this in your able hands.

Here is the `.htaccess` file itself:

```
RewriteEngine On
RewriteBase /ns/credentials/
AddType application/ld+json .jsonld
AddType text/turtle .ttl


# The v2 of the core VCDM JSON-LD context:
RewriteRule ^v2$ https://w3c.github.io/vc-data-model/contexts/credentials/v2 [E=json,P]

# The vocabulary referred to via a @vocab statement at the end of the core VCDM JSON-LD context
# This was removed, see https://github.com/w3c/vc-data-model/pull/1536
# RewriteRule ^issuer-dependent$ https://w3c.github.io/vc-data-model/vocab/credentials/v2/issuer-dependent.html [P]

# The v2 of the example JSON-LD context, used for the examples in the VCDM specification
RewriteRule ^examples/v2$ https://w3c.github.io/vc-data-model/contexts/credentials/examples/v2 [E=json,P]
# The v2 example context file is just a @vocab statement to ns/credentials/examples
RewriteRule ^examples$ https://w3c.github.io/vc-data-model/vocab/credentials/v2/examples.html [P]
RewriteRule ^examples#$ https://w3c.github.io/vc-data-model/vocab/credentials/v2/examples.html [P]

# The v2 of the undefined term JSON-LD context, used for the undefined terms in the VCDM specification
RewriteRule ^undefined-term/v2$ https://w3c.github.io/vc-data-model/contexts/credentials/undefined-term/v2 [E=json,P]
# The v2 example context file is just a @vocab statement to ns/credentials/examples
RewriteRule ^undefined-term$ https://w3c.github.io/vc-data-model/vocab/credentials/v2/undefined-term.html [P]
RewriteRule ^undefined-term#$ https://w3c.github.io/vc-data-model/vocab/credentials/v2/undefined-term.html [P]

#### Status list additions
RewriteRule ^status$ https://www.w3.org/ns/credentials/status/ [P]

Header set Content-Type application/ld+json env=json
```

### CODE REVIEW

**Review by decentralgabe — APPROVED** [review]

File: vocab/credentials/v2/issuer-dependent.html [hunk 1]
```diff
@@ -66,7 +54,7 @@
...
+            The <a href="https://www.w3.org/TR/vc-data-model-2.0/#base-context">base context</a> of a Verifiable Credential, as defined in [[vc-data-model-2.0]], provides a mapping from the standard terms used in credentials to the vocabulary items defined by the Verifiable Credentials Working Group through different vocabularies, such as the <a href="https://www.w3.org/2018/credentials/">Verifiable Credentials Vocabulary</a>. Applications using additional, user-dependent vocabularies of their own are encouraged to create their own formal vocabularies, and corresponding JSON-LD context definitions, to ensure the interoperability of the credentials they produce and/or consume.
```
Comment (TallTed) [inline vocab/credentials/v2/issuer-dependent.html:57]:
```suggestion
            The <a href="https://www.w3.org/TR/vc-data-model-2.0/#base-context">base context</a> of a Verifiable Credential, as defined in [[[vc-data-model-2.0]]] [[vc-data-model-2.0]], provides a mapping from the standard terms used in credentials to the vocabulary items defined by the Verifiable Credentials Working Group through different vocabularies, such as the <a href="https://www.w3.org/2018/credentials/">Verifiable Credentials Vocabulary</a>. Applications using additional, user-dependent vocabularies of their own are encouraged to create their own formal vocabularies, and corresponding JSON-LD context definitions, to ensure the interoperability of the credentials they produce and/or consume.
```

**Review by TallTed — CHANGES_REQUESTED** [review]
Tweak citation reference

File: vocab/credentials/v2/issuer-dependent.html [hunk 2]
```diff
@@ -12,18 +12,6 @@
     </script>
     <script class="remove">
```
Comment (msporny) [inline vocab/credentials/v2/issuer-dependent.html:13]: This whole file should be deleted, right? 

"issuer-dependent" is no longer a thing this WG defines.

File: vocab/credentials/v2/undefined-term.html [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+            Comments regarding this document are welcome. Please file issues
+            directly on <a href="https://github.com/w3c/vc-data-model/issues/">GitHub</a>, or send them to
+            <a href="mailto:public-vc-comments@w3.org">public-vc-comments@w3.org</a>
+            (<a href="mailto:public-vc-comments-request@w3.org?subject=subscribe">subscribe</a>,
+            <a href="https://lists.w3.org/Archives/Public/public-vc-comments/">archives</a>).    
+        </p>
+    </section>
+    <section id="conformance"></section>
+    <section>
+        <h2>Issuer Dependent Terms</h2>
```
Comment (msporny) [inline vocab/credentials/v2/undefined-term.html:55]:
```suggestion
        <h2>Undefined Terms</h2>
```

File: vocab/credentials/v2/undefined-term.html [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        <p>
+            The <a href="https://www.w3.org/TR/vc-data-model-2.0/#base-context">base context</a> of a Verifiable Credential, as defined in [[vc-data-model-2.0]], provides a mapping from the standard terms used in credentials to the vocabulary items defined by the Verifiable Credentials Working Group through different vocabularies, such as the <a href="https://www.w3.org/2018/credentials/">Verifiable Credentials Vocabulary</a>. Applications using additional, user-dependent vocabularies of their own are encouraged to create their own formal vocabularies, and corresponding JSON-LD context definitions, to ensure the interoperability of the credentials they produce and/or consume.
```
Comment (msporny) [inline vocab/credentials/v2/undefined-term.html:57]:
```suggestion
            The <a href="https://www.w3.org/TR/vc-data-model-2.0/#base-context">base context</a> of a Verifiable Credential, as defined in [[vc-data-model-2.0]], provides a mapping from the standard terms used in credentials to the vocabulary items defined by the Verifiable Credentials Working Group through different vocabularies, such as the <a href="https://www.w3.org/2018/credentials/">Verifiable Credentials Vocabulary</a>. Applications using additional, application-specific terms of their own are encouraged to create their own formal vocabularies, and corresponding JSON-LD context definitions, to ensure the interoperability of the credentials they produce and/or consume.
```

File: vocab/credentials/v2/undefined-term.html [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        </p>
+        <p>
+            There may be cases, however, when the creation of such formal vocabularies and context definitions is too much of a burden, such as for closed applications, experiments, or examples. The <a href="https://www.w3.org/ns/credentials/undefined-term/v2">context</a> has been constructed in such a way that issuers may use new, hitherto undefined terms by adding it to the Verifiable Credential, and these terms will be mapped onto URLs of the form <code>https://www.w3.org/ns/credentials/undefined-term#TERM</code>. If a conforming document does not use JSON-LD Contexts that define all terms used, it must include the `https://www.w3.org/ns/credentials/undefined-terms/v2` as the last value in the `@context` property.
```
Comment (msporny) [inline vocab/credentials/v2/undefined-term.html:60]: There is lowercase normative language such as `may` and `must` in here that we might not want to use... people might get confused by whether this is normative language or not (it's not, the main spec says these things, IIRC).

**Review by msporny — CHANGES_REQUESTED** [review]
Overall LGTM, some minor nits before we merge.

File: vocab/credentials/v2/issuer-dependent.html (hunk 2 above)
Comment (iherman) [reply vocab/credentials/v2/issuer-dependent.html:13]: I was not sure about that. But yes, that is what I expected to do.

Removed in https://github.com/w3c/vc-data-model/pull/1536/commits/adaffa8248d0f560e576954449944bf7f85b12a3.

I have also removed it from the .htaccess file above.

File: vocab/credentials/v2/undefined-term.html (hunk 5 above)
Comment (iherman) [reply vocab/credentials/v2/undefined-term.html:60]: I think lower case must/may is o.k. here. This document is not a Recommendation, and a upper case version would possibly raise eyebrows.

(Besides, I believe respec might complain...)

Comment (iherman) [reply vocab/credentials/v2/undefined-term.html]: Made the change in https://github.com/w3c/vc-data-model/pull/1536/commits/c41517aa88e7888f73daac698481ff6226d2c6ec

File: vocab/credentials/v2/undefined-term.html [hunk 6]
```diff
@@ -64,13 +52,21 @@
...
+          There may be cases, however, when the creation of such formal vocabularies and context definitions is too much of a
+          burden, such as for closed applications, experiments, or examples. The <a
+            href="https://www.w3.org/ns/credentials/undefined-term/v2">context</a> has been constructed in such a way that issuers
+          may use new, hitherto undefined terms by adding it to the Verifiable Credential, and these terms will be mapped onto
+          URLs of the form <code>https://www.w3.org/ns/credentials/undefined-term#TERM</code>. If a conforming document does not
+          use JSON-LD Contexts that define all terms used, it must include the
+          `https://www.w3.org/ns/credentials/undefined-terms/v2` as the last value in the `@context` property.
```
Comment (TallTed) [inline vocab/credentials/v2/undefined-term.html:67]:
```suggestion
There may be cases, however, where the creation of such formal vocabularies and
context definitions is too much of a burden, such as for closed applications,
experiments, or examples. The <a
href="https://www.w3.org/ns/credentials/undefined-term/v2">Undefined Terms
context</a> has been constructed in such a way that issuers may use new,
hitherto undefined terms by adding this context to their Verifiable Credentials,
which will cause such terms to be mapped onto URLs of the form
`https://www.w3.org/ns/credentials/undefined-term#TERM`. If a conforming
document does not use JSON-LD Contexts that define all terms used, it is
required to include `https://www.w3.org/ns/credentials/undefined-terms/v2` as
the last value in the `@context` property.
```

**Review by TallTed — CHANGES_REQUESTED** [review]
Editorial fixups

**Review by selfissued — CHANGES_REQUESTED** [review]
I believe this is a mistake. We would be better off with `@vocab` in the main context so there are no undefined terms.

**Comment by msporny** [comment]
> We would be better off with @vocab in the main context so there are no undefined terms.

That ship has sailed. PR #1520, removing `@vocab` from the base context, was merged after considerable debate and with WG consensus.

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/2017/vc/WG/Meetings/Minutes/2024-07-17-vcwg#section1-3) on 2024-07-17

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

#### 1.3. Publish undefined-terms context URI (issue vc-data-model#1534)

_See github issue [vc-data-model#1534](https://github.com/w3c/vc-data-model/issues/1534)._




_See github pull request [vc-data-model#1536](https://github.com/w3c/vc-data-model/pull/1536)._




**Manu Sporny:** for next item Ivan has raised a PR, once those three are done, no more issues on VCDM.  
… only thing waiting on then is for test suites to demonstrate multiple independent implementations for every feature in specification.  

</details>

**Comment by iherman** [comment]
@msporny can this PR be merged (and the .htaccess file updated)?

**Review by TallTed — APPROVED** [review]

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/2017/vc/WG/Meetings/Minutes/2024-08-07-vcwg#section1-1) on 2024-08-07

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

#### 1.1. Added the new vocabulary file for undefined terms (pr vc-data-model#1536)

_See github pull request [vc-data-model#1536](https://github.com/w3c/vc-data-model/pull/1536)._




**Brent Zundel:** We have a PR we should look at.  
… There are a number of requests for changes here, Ivan can you give us a summary of what the PR does?  
… I encourage folks who would like changes to also jump in.  

**Ivan Herman:** Just to say what you said here a little differently. There were a number of comments, so they have all been incorporated, so as far as I know, no more open comments.  
… What it does is -- following up on the PR(s) which changed the place for the `@vocab` term and now there is a separate undefined term that has to be done properly and this PR handles that.  
… There is one part of the PR that isn't in there which is for htaccess file changes I have to do externally.  
… This will handle redirections, etc. it's not on github, won't be subject to PR, but I put a verbatim copy of the .htaccess file.  
… There were some small comments incorporated on that as well, as far as I am concerned this can be merged and then I will change the .htaccess file after that.  

**Brent Zundel:** Thanks for that summary, Ivan.  
… To add a little bit more context, I believe three weeks ago this group came to consensus on and merged two PRs to make changes to add the undefined terms context and remove the `@vocab` from the core context.  
… This PR just does administrative work to finish up those things that were done nearly a month ago.  

**Ivan Herman:** That is correct.  

**Brent Zundel:** Ted and Mike Jones, your reviews are still showing requests for changes, Ted your changes have been merged, Mike if you'd like to jump on the queue you can do so.  

**Michael Jones:** Alright, I still think it's a mistake but I think the ship has sailed and I'll approve it.  

**Brent Zundel:** Thank you Mike, I appreciate that.  
… Ok, if there are no more comments on this PR, then we can look briefly at our set of issues.  

> *Brent Zundel:* [https://github.com/w3c/vc-data-model/issues?q=is%3Aissue+is%3Aopen+-label%3Afuture+sort%3Aupdated-asc](https://github.com/w3c/vc-data-model/issues?q=is%3Aissue+is%3Aopen+-label%3Afuture+sort%3Aupdated-asc).

**Brent Zundel:** There are two remaining issues on the VCDM.  
… I think the first issue will be addressed by the PR we just looked at.  

</details>

**Comment by iherman** [comment]
Merging per https://www.w3.org/2017/vc/WG/Meetings/Minutes/2024-08-07-vcwg#section1-1. The `.htaccess` file has also been updated on the W3C server.

**Pull request merged**